test-suite/start-test-environment.sh

```diff
@@ -40,11 +40,26 @@ docker-compose -f docker-compose.yml up -d activemq redis dynamodb ps_db outboun
 docker-compose -f docker-compose.yml up db_migration
 docker-compose -f docker-compose.yml rm -f db_migration
 
-cd docker/snomed-database-loader
-./load_release-postgresql.sh ${SNOMED_FILE_LOCATION}
-cd ../..
-cd docker/snomed-immunization-loader
-./load_immunization_codes.sh
-cd ../..
+snomedexists=$(docker-compose exec -u postgres ps_db psql -d patient_switching -AXqtc "SELECT EXISTS (SELECT 1 FROM information_schema.schemata WHERE schema_name='snomedct')")
+
+if [ $snomedexists != 't' ]
+then
+  cd docker/snomed-database-loader
+  ./load_release-postgresql.sh ${SNOMED_FILE_LOCATION}
+  cd ../..
+else
+  echo "SNOMED CT schema already exists, skipping loading"
+fi
+
+immunizationcodesexist=$(docker-compose exec -u postgres ps_db psql -d patient_switching -AXqtc "SELECT COUNT(*) FROM snomedct.immunization_codes")
+
+if [ $immunizationcodesexist == '0' ]
+then
+  cd docker/snomed-immunization-loader
```

Comment:
When I run the script, it prints out the steps its performing. Does it not do that on Windows?

Reply:
Yes, it does but I think as there are a lot of log entries(what it starts pumping the data) sometimes it is difficult to recognize what is going on (especially for somebody who possibly just starts working with it) hence a small log entry before the data pumping would be still beneficial I think...

Reply:
If there are lots of log entries should the original ones be suppressed too? Otherwise I'm unsure how adding more log entries would help that?

Reply:
obviously, it is not critical as it is just the log entry.... but there are a lot of log entries because we are running it in debug "set -x  -e" mode however if that is disabled (even with enabled one I still think there is some value in it)  and there are less log entries being added then such an additional log entry (about what is about to happen next) is more informative and helpful in my humble opinion but it might be only my subjective view... I am leaving it with you to decide...